test/helpers/patient_query_helper_test.rb

```diff
@@ -438,4 +438,362 @@ class PatientQueryHelperTest < ActionView::TestCase
     filtered_patients_array = [patient_1, patient_6]
     assert_equal filtered_patients_array.map { |p| p[:id] }, filtered_patients.pluck(:id)
   end
+
+  test 'advanced filter vaccination single filter option vaccine group' do
+    Patient.destroy_all
+    user = create(:public_health_enroller_user)
+    patient_1 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_1, group_name: 'COVID-19')
+    create(:vaccine, patient: patient_1, group_name: 'COVID-19')
+    patient_2 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_2, group_name: 'COVID-19')
+    create(:patient, creator: user)
+
+    patients = Patient.all
+    filters = [{ filterOption: {}, value: [{ name: 'vaccine-group', value: 'COVID-19' }] }]
+    filters[0][:filterOption]['name'] = 'vaccination'
+    tz_offset = 300
+    filtered_patients = advanced_filter(patients, filters, tz_offset)
+    filtered_patients_array = [patient_1, patient_2]
+    assert_equal filtered_patients_array.map { |p| p[:id] }, filtered_patients.pluck(:id)
+  end
+
+  test 'advanced filter vaccination single filter option product name' do
+    Patient.destroy_all
+    user = create(:public_health_enroller_user)
+    patient_1 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_1, product_name: 'Moderna COVID-19 Vaccine')
+    create(:vaccine, patient: patient_1, product_name: 'Pfizer-BioNTech COVID-19 Vaccine')
+    patient_2 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_2, product_name: 'Janssen (J&J) COVID-19 Vaccine')
+    patient_3 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_3, product_name: 'Moderna COVID-19 Vaccine')
+    patient_4 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_4, product_name: 'Unknown')
+    patient_5 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_5, product_name: 'Pfizer-BioNTech COVID-19 Vaccine')
+    create(:vaccine, patient: patient_5, product_name: 'Pfizer-BioNTech COVID-19 Vaccine')
+    patient_6 = create(:patient, creator: user)
+    create(:vaccine, patient: patient_6, product_name: 'Moderna COVID-19 Vaccine')
+    create(:vaccine, patient: patient_6, product_name: 'Moderna COVID-19 Vaccine')
+    create(:patient, creator: user)
```

Comment:
There's a lot of building up of test data in these. It would be ideal to use YAML fixtures instead ([here](https://github.com/SaraAlert/SaraAlert/blob/7b603d77e95bd3ff75536e1e1b3b4a8c781c062a/test/fixtures/vaccines.yml)). That way you'd only have to define a single set of data to test against rather than an entirely new one for each test, and it'd be a lot more maintainable.

Reply:
@tstrass I would actually prefer these patients/vaccines to be defined here because the test fixtures are typically more for the system tests

Reply:
Ah right. In that case, what would you think about instead defining a setup method in this test file to build up a single set of test data that is reused by all of these filter tests?

Reply:
I think that would make sense if the different test cases reuse the same patients/vaccines/labs/etc, but each of these records are usually setup specifically to test each of these filters

Reply:
a lot of these are different, the look the same but aren't quite the same, so it still would be a lot of data. not sure it would be worth the refactor but open to either option

It's also easier to debug this way because you can see everything right in the test

Reply:
Do they need to all be setting up different test data though? Can't you just essentially combine them?

We're just testing different combinations of filters, and that can just as easily be done on a single set of data. Like to test the single filter for dose number, you don't need a set of vaccines that _only_ has dose numbers.

Reply:
That's true, but there's still quite a few different vaccines that would need to be uses (20-30). not all of them are used in every test and are associated with different patients throughout.  my preference would be to leave as is because I think it's clearer to track down where things are failing and avoids having to rewrite the whole test suite

Reply:
Ok, I'll close this thread since it's already written this way, but only for that reason. Consolidating to a single set of data would in no way make it harder to track down where things are failing.

In the future, it is preferable to reuse test data across similar tests. It's just more maintainable; if something changes in the vaccine model or in the way vaccines are created, you're updating the test data in a single place rather than for every single test, and it's just significantly fewer lines of code.